pkg/ee/default-policy-controller/controller.go

```diff
@@ -0,0 +1,399 @@
+	return reconciling.ReconcilePolicyBindings(ctx, reconcilers, namespace, r.Client)
+}
+
+func PolicyBindingReconciler(logger *zap.SugaredLogger, template kubermaticv1.PolicyTemplate, existingBinding *kubermaticv1.PolicyBinding, cluster *kubermaticv1.Cluster) reconciling.NamedPolicyBindingReconcilerFactory {
+	return func() (string, reconciling.PolicyBindingReconciler) {
+		bindingName := template.Name
+		return bindingName, func(binding *kubermaticv1.PolicyBinding) (*kubermaticv1.PolicyBinding, error) {
+			delete(template.Annotations, corev1.LastAppliedConfigAnnotation)
+
+			annotations := template.Annotations
+			if annotations == nil {
+				annotations = make(map[string]string)
+			}
+
+			if template.Spec.Enforced {
+				annotations[kubermaticv1.AnnotationPolicyEnforced] = strconv.FormatBool(true)
+			}
+			if template.Spec.Default {
+				annotations[kubermaticv1.AnnotationPolicyDefault] = strconv.FormatBool(true)
+			}
+
+			// Copy labels from the template if any
+			if len(template.Labels) > 0 {
+				if binding.Labels == nil {
+					binding.Labels = make(map[string]string)
+				}
+				for k, v := range template.Labels {
+					binding.Labels[k] = v
+				}
+			}
+
+			binding.Annotations = annotations
+			binding.Spec = kubermaticv1.PolicyBindingSpec{
+				PolicyTemplateRef: corev1.ObjectReference{
+					Name: template.Name,
+				},
+			}
+
+			logger.Infof("Reconciling PolicyBinding %s/%s", binding.Namespace, binding.Name)
+			return binding, nil
+		}
+	}
+}
+
+// isClusterTargeted checks if the PolicyTemplate targets the given cluster.
+func isClusterTargeted(cluster *kubermaticv1.Cluster, template *kubermaticv1.PolicyTemplate) bool {
+	// If no target is specified, we check the visibility
+	if template.Spec.Target == nil {
+		// Global policies apply to all clusters
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityGlobal {
+			return true
+		}
+
+		// Project policies apply to clusters in the same project
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityProject &&
+			template.Spec.ProjectID != "" &&
+			cluster.Labels[kubermaticv1.ProjectIDLabelKey] == template.Spec.ProjectID {
+			return true
+		}
+
+		return false
+	}
+
+	// Check project selector if specified
+	if template.Spec.Target.ProjectSelector != nil && cluster.Labels[kubermaticv1.ProjectIDLabelKey] != "" {
+		selector, err := metav1.LabelSelectorAsSelector(template.Spec.Target.ProjectSelector)
+		if err != nil {
+			return false
+		}
+
+		projectLabels := map[string]string{kubermaticv1.ProjectIDLabelKey: cluster.Labels[kubermaticv1.ProjectIDLabelKey]}
+		if !selector.Matches(labels.Set(projectLabels)) {
+			return false
+		}
+	}
+
+	// Check cluster selector if specified
+	if template.Spec.Target.ClusterSelector != nil {
+		selector, err := metav1.LabelSelectorAsSelector(template.Spec.Target.ClusterSelector)
+		if err != nil {
+			return false
+		}
+
+		if !selector.Matches(labels.Set(cluster.Labels)) {
+			return false
+		}
+	}
```

Comment:
```suggestion
	return selector.Matches(labels.Set(cluster.Labels))
```

Reply:
This would be either invalid or change behaviour :grin:, depending on whether you really meant to include line 321 in your selection.

In the project selector check above, we also do not stop if the selector matches, so I can see both if blocks being formatted the same way, if a fallthrough is intended.